### ISSUE TITLE
405 Methods not Allowed does not work

### ISSUE THREAD

**Issue by bakura10** [zendframework/zend-expressive#34]
Assuming the following config: 

```php
'services' => [
        'config' => [
            'routes' => [
                [
                    'path' => '/',
                    'middleware' => 'Application\HelloWorld',
                    'allowed_methods' => [ 'POST' ],
                ],
            ],
        ]
    ],
```

Hitting "/" returns a 404, while expected result is 405. It seems the logic handled for "isMethodFailure" is wrong.

**Comment by weierophinney** [comment]
@bakura10 — This will be corrected with #28.

`Zend\Stratigility\FinalHandler` essentially does the following:

- If an `$error` is provided, it returns a 500 response.
- If no `$error` is present:
  - If the response provided when invoked is a different instance from the one provided at instantiation (which should be the case after #28 is merged), it will return that response.
  - If the response body size differs from when `FinalHandler` was instantiated, it returns the response.
  - Otherwise, it returns a 404.

As noted, #28 should fix the situation you're reporting. I'll close this issue when that PR is merged (I'm having @ezimuel  review it first).

**Comment by ojhaujjwal** [comment]
#28 is merged now
